PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tempGraph/DrawGraph.java

```diff
@@ -113,18 +119,68 @@ private void drawEdges(Pane pane, HashMap<Integer, Bubble> bubbles, HashMap<Inte
 			Circle fromCircle = circles.get(id);
 			for (Integer outLink : from.getOutLinks()) {
 				Circle toCircle = circles.get(outLink);
-				Line edge = new Line();
-				edge.startXProperty().bind(fromCircle.centerXProperty());
-				edge.startYProperty().bind(fromCircle.centerYProperty());
-				edge.endXProperty().bind(toCircle.centerXProperty());
-				edge.endYProperty().bind(toCircle.centerYProperty());
-				pane.getChildren().add(edge);
-				edge.toBack();
+				QuadCurve curve = new QuadCurve();
+
+                curve.setFill(Color.TRANSPARENT);
+                curve.setStroke(Color.BLACK);
+                curve.setStrokeWidth(2.0d);
+
+                fromCircle.centerXProperty().addListener((observable, oldValue, newValue) -> {
+                    curve.startXProperty().setValue(newValue);
+                    bendCurve(curve, fromCircle, toCircle);
+                });
+                fromCircle.centerYProperty().addListener((observable, oldValue, newValue) -> {
+                    curve.startYProperty().setValue(newValue);
+                    bendCurve(curve, fromCircle, toCircle);
+                });
+                toCircle.centerXProperty().addListener((observable, oldValue, newValue) -> {
+                    curve.endXProperty().setValue(newValue);
+                    bendCurve(curve, fromCircle, toCircle);
+                });
+                toCircle.centerYProperty().addListener((observable, oldValue, newValue) -> {
+                    curve.endYProperty().setValue(newValue);
+                    bendCurve(curve, fromCircle, toCircle);
+                });
+
+                pane.getChildren().add(curve);
+				curve.toBack();
 //				edge.setSmooth(true);
 			}
 		});
 	}
 
+    private static void bendCurve(QuadCurve curve, Circle from, Circle to) {
+        Point2D perpP;
+        if (from.getCenterY() > to.getCenterY()) {
+            perpP = getPerpendicularPoint(from.getCenterX(), from.getCenterY(), to.getCenterX(), to.getCenterY(), 20.0d);
+        } else {
+            perpP = getPerpendicularPoint(to.getCenterX(), to.getCenterY(), from.getCenterX(), from.getCenterY(), 20.0d);
+        }
+
+        curve.setControlX(perpP.getX());
+        curve.setControlY(perpP.getY());
+    }
+
+    /**
+     *
+     * Not that 4(!) objects are created for a single invocation. This is considerably slower than
+     * a more efficient way of handling thing. (atleast until the JIT kicks in it seems)
```

Comment:
Obviously anything is slower than a more efficient way of doing the same thing. :+1: